lib/spack/spack/package.py

```diff
@@ -983,28 +977,42 @@ def module(self):
                           fromlist=[self.__class__.__name__])
 
 
-    def setup_dependent_environment(self, module, spec, dependent_spec):
-        """Called before the install() method of dependents.
+    def environment_modifications(self, dependent_spec):
```

Comment:
I like the first for this stuff b/c it's less typing for the package author (who ideally doesn't have to know Python or OO really well).  The second would make sense to me if it was a more complicated framework and they could ever return something other than an `EnvironmentModifications` but here it just seems like extra boilerplate.